src/imagecapturecore.cs

```diff
@@ -0,0 +1,789 @@
+		ICScannerDocumentType DocumentType { get; set; }
+	
+		[Export ("documentSize")]
+		CGSize DocumentSize { get; }
+	}
+	
+	[BaseType (typeof(ICScannerFunctionalUnit))]
+	interface ICScannerFunctionalUnitDocumentFeeder {
+
+		[Export ("supportedDocumentTypes")]
+		NSIndexSet SupportedDocumentTypes { get; }
+	
+		[Export ("documentType", ArgumentSemantic.Assign)]
+		ICScannerDocumentType DocumentType { get; set; }
+	
+		[Export ("documentSize")]
+		CGSize DocumentSize { get; }
+	
+		[Export ("supportsDuplexScanning")]
+		bool SupportsDuplexScanning { get; }
+	
+		[Export ("duplexScanningEnabled")]
+		bool DuplexScanningEnabled { get; set; }
+	
+		[Export ("documentLoaded")]
+		bool DocumentLoaded { get; }
+	
+		[Export ("oddPageOrientation", ArgumentSemantic.Assign)]
+		ICExifOrientationType OddPageOrientation { get; set; }
+	
+		[Export ("evenPageOrientation", ArgumentSemantic.Assign)]
+		ICExifOrientationType EvenPageOrientation { get; set; }
+	
+		[Export ("reverseFeederPageOrder")]
+		bool ReverseFeederPageOrder { get; }
+	}
+	
+	[BaseType (typeof(NSObject))]
+	interface ICScannerBandData {
+
+		[Export ("fullImageWidth")]
+		nuint FullImageWidth { get; }
+	
+		[Export ("fullImageHeight")]
+		nuint FullImageHeight { get; }
+	
+		[Export ("bitsPerPixel")]
+		nuint BitsPerPixel { get; }
+	
+		[Export ("bitsPerComponent")]
+		nuint BitsPerComponent { get; }
+	
+		[Export ("numComponents")]
+		nuint NumComponents { get; }
+	
+		[Export ("bigEndian")]
+		bool BigEndian { [Bind ("isBigEndian")] get; }
+	
+		[Export ("pixelDataType")]
+		ICScannerPixelDataType PixelDataType { get; }
+	
+		[NullAllowed, Export ("colorSyncProfilePath", ArgumentSemantic.Retain)]
+		string ColorSyncProfilePath { get; }
+	
+		[Export ("bytesPerRow")]
+		nuint BytesPerRow { get; }
+	
+		[Export ("dataStartRow")]
+		nuint DataStartRow { get; }
+	
+		[Export ("dataNumRows")]
+		nuint DataNumRows { get; }
+	
+		[Export ("dataSize")]
+		nuint DataSize { get; }
+	
+		[NullAllowed, Export ("dataBuffer", ArgumentSemantic.Retain)]
+		NSData DataBuffer { get; }
+	}
+
+	interface IICScannerDeviceDelegate {}
+
+	[Protocol, Model (AutoGeneratedName = true)]
+	[BaseType (typeof (NSObject))]
+	interface ICScannerDeviceDelegate : ICDeviceDelegate {
+
+		[Export ("scannerDeviceDidBecomeAvailable:")]
+		void DidBecomeAvailable (ICScannerDevice scanner);
+	
+		[Export ("scannerDevice:didSelectFunctionalUnit:error:")]
+		void DidSelectFunctionalUnit (ICScannerDevice scanner, ICScannerFunctionalUnit functionalUnit, [NullAllowed] NSError error);
+	
+		[Deprecated (PlatformName.MacOSX, 10, 7)]
+		[Export ("scannerDevice:didScanToURL:data:")]
+		void DidScanToUrl (ICScannerDevice scanner, NSUrl url, NSData data);
+	
+		[Export ("scannerDevice:didScanToURL:")]
+		void DidScanToUrl (ICScannerDevice scanner, NSUrl url);
+	
+		[Export ("scannerDevice:didScanToBandData:")]
+		void DidScanToBandData (ICScannerDevice scanner, ICScannerBandData data);
```

Comment:
Same probably better to go `DidScan`. Thoughts?

Reply:
same commet as Chamons, better to be explicit on the callback names.

Reply:
Agree, If all of them were `DidScanToUrl` it would make sense to go `DidScan` but we have two different `DidScanToBandData` and `DidScanToUrl` I think they are fine as is.